### PR TITLE
Compile and import common into discord-bot

### PR DESCRIPTION
**Pull request by IanPhilips**
I can't seem to get this to work @mqp - I copied your `package.json` scripts from the replicator. Do you have any idea what I'm doing wrong? Perhaps some important difference between our `tsconfig.json`'s?
When running the transpiled files they can't access the exports even though the necessary files are all there in the dist/common/lib/utils and supabase folders. 

Screenshot 2023-02-27 at 7 29 09 AM

### CODE REVIEW

File: discord-bot/Dockerfile [hunk 1]
```diff
@@ -3,10 +3,10 @@
...
+COPY dist/package.json dist/yarn.lock ./
 RUN yarn install --frozen-lockfile --production
 
-# Will have to rewrite this if we bring in outside dependencies
-COPY . ./
+# then copy over typescript payload
+COPY dist ./
 
-CMD ["node", "lib/index.js"]
+CMD ["node", "discord-bot/lib/index.js"]
```
Comment (mqp) [inline discord-bot/Dockerfile:12]: To my mind, the sane way to do this is to say, "the directory structure of dist/ is exactly the same as the directory structure of manifold/". Anything else is confusing. So I would have made the discord bot code go in `dist/backend/discord-bot`, not `dist/discord-bot`.

File: discord-bot/package.json [hunk 2]
```diff
@@ -3,9 +3,15 @@
...
-    "dev": "ENVIRONMENT=DEV nodemon --watch src --exec ts-node src/index.ts"
+    "start": "yarn build:watch \"ENVIRONMENT=PROD nodemon --watch lib --exec node lib/index.js\"",
+    "dev": "yarn build:watch \"ENVIRONMENT=DEV nodemon --watch lib --exec node lib/index.js\"",
+    "alias": "tsc-alias",
+    "compile": "tsc -b",
+    "build:watch": "concurrently --kill-others \"tsc -w\" \"tsc-alias -w\"",
+    "build": "yarn compile && yarn --cwd=../common alias && yarn alias && yarn dist",
+    "dist": "yarn dist:prepare && yarn dist:copy",
+    "dist:prepare": "rm -rf dist && mkdir -p dist/common/lib  dist/discord-bot/lib",
+    "dist:copy": "cp -R ../common/lib/* dist/common/lib && cp -R ./lib/* dist/discord-bot/lib && cp ../yarn.lock dist && cp package.json dist"
```
Comment (mqp) [inline discord-bot/package.json:14]: As stated above, I would copy the code to `dist/backend/discord-bot/lib`.

File: discord-bot/tsconfig.json [hunk 3]
```diff
@@ -24,7 +24,9 @@
             "path": "../common"
         },
     ],
-
+    "tsc-alias": {
+        "resolveFullPaths": true
+    },
```
Comment (mqp) [inline discord-bot/tsconfig.json:29]: Not sure why this is necessary.

**Comment by IanPhilips** [comment]
For posterity: https://discord.com/channels/915138780216823849/959728429295616061/1080258732971270165

File: discord-bot/Dockerfile (hunk 1 above)
Comment (IanPhilips) [reply discord-bot/Dockerfile]: At the time, `discord-bot` wasn't in the `backend` directory, so this wasn't right, but now it is and this is appropriate advice!

**Comment by IanPhilips** [comment]
Switching the `common` dir to a `es2022` module was the thing that fixed my error, thanks @mqp!

File: discord-bot/tsconfig.json (hunk 3 above)
Comment (IanPhilips) [reply discord-bot/tsconfig.json:28]: This allows me to import like so: `import { randomString } from 'common/util/random'` and `import { config } from 'discord-bot/constants/config'` without adding the `.js` extension normally required in es modules

**Pull request merged**